relay_benchmark_test.go

```diff
@@ -223,3 +225,20 @@ func BenchmarkRelay2Servers5Clients4k(b *testing.B) {
 	p.servers = 2
 	benchmarkRelay(b, p)
 }
+
+func BenchmarkRelayWithAppends(b *testing.B) {
+	n, err := strconv.Atoi(os.Getenv("BENCH_RELAY_APPENDS_N"))
+	require.NoError(b, err)
+
+	p := defaultParams()
+	for i := 0; i < n; i++ {
```

Comment:
I think the flexibility is probably not needed, we don't really need to test the incremental cost of each append (e.g., 6 --> 7 won't provide much signal). I think the cases we want are:
 - 0 appends (covered below with `WithoutAppends`)
 - 1 append (what we expect to use for token injection)
 - 2 appends (basically the border line for the backing array capacity)
then we probably just need some higher values like 5 and 10 to see if the cost increases linearly for each append or not
so basically just `[1, 2, 5, 10]`

Reply:
really good points. new results:
```
 0    9948	    114703 ns/op	   8.93 MB/s	     617 B/op	      12 allocs/op
 1    9166	    120177 ns/op	   8.52 MB/s	    1122 B/op	      19 allocs/op
 2    9906	    116765 ns/op	   8.77 MB/s	    1101 B/op	      19 allocs/op
 5   10011	    118211 ns/op	   8.66 MB/s	    1692 B/op	      21 allocs/op
10    9634	    134943 ns/op	   7.59 MB/s	    2572 B/op	      22 allocs/op
```